marketplacetests/marketplace/app.py

```diff
@@ -104,6 +104,7 @@ def search(self, term):
 
     def set_region(self, region):
         # go to the :debug page
+        self.wait_for_element_displayed(*self._search_locator)
```

Comment:
There may be other places this could help. I will look through the rest of the code. As for your second question, I don't know. There have been a number of times that we've seen `send_keys` fail silently, so this actually doesn't surprise me at all.